src/coreclr/jit/llvm.cpp

```diff
@@ -1652,33 +1708,43 @@ Value* Llvm::getLocalVarAddress(GenTreeLclVar* lclVar) {
     return _builder.CreateGEP(_function->getArg(0), _builder.getInt32(varOffset), "lclVar");
 }
 
+bool Llvm::isLlvmFrameLocal(LclVarDsc* varDsc)
+{
+    assert(canStoreLocalOnLlvmStack(varDsc) && (_compiler->fgSsaPassesCompleted >= 1));
+    return !varDsc->lvInSsa;
+}
+
 void Llvm::storeLocalVar(GenTreeLclVar* lclVar)
 {
-    if (lclVar->gtFlags & GTF_VAR_DEF)
+    Value* localValue = getGenTreeValue(lclVar->gtGetOp1());
+
+    unsigned lclNum = lclVar->GetLclNum();
+    Value* allocaValue = m_allocas[lclNum];
+    LclVarDsc* varDsc  = _compiler->lvaGetDesc(lclVar);
+
+    if (isLlvmFrameLocal(varDsc))
+    {
+        _builder.CreateStore(localValue, castIfNecessary(allocaValue, localValue->getType()->getPointerTo()));
```

Comment:
```suggestion
    LclVarDsc* varDsc = _compiler->lvaGetDesc(lclVar);

    if (isLlvmFrameLocal(varDsc))
    {
        Value* lclAddressValue = m_allocas[lclNum];
        _builder.CreateStore(localValue, castIfNecessary(lclAddressValue, localValue->getType()->getPointerTo()));
```
(Moved it to the scope where it is used)